package/src/dep/mod.rs

```diff
@@ -0,0 +1,71 @@
+use alloc::string::String;
+
+use serde::{Deserialize, Serialize};
+
+use super::{de, se};
+use crate::Digest;
+
+pub(crate) mod resolver;
+
+/// A system library identifier
+#[derive(
+    Debug, Copy, Clone, PartialEq, Eq, serde_repr::Serialize_repr, serde_repr::Deserialize_repr,
+)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+#[repr(u8)]
+pub enum SystemLibraryId {
+    /// The standard library
+    Stdlib,
+    /// The base library
+    Miden,
+}
+
+impl core::str::FromStr for SystemLibraryId {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            // Compiler uses "std" and "base" to identify the standard and base libraries
+            // respectively. We also accept "stdlib" and "miden" as aliases for these libraries.
+            "std" | "stdlib" => Ok(Self::Stdlib),
+            "base" | "miden" => Ok(Self::Miden),
+            _ => Err(()),
+        }
+    }
+}
+
+/// The name of a dependency
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub enum DependencyName {
+    /// The dependency is a system library
+    System(SystemLibraryId),
+    /// The dependency is a user library with the given name
+    User(String),
+}
```

Comment:
Related to the previous comment, maybe this could be just `DependencyName(String)`?

Reply:
Done.